### ISSUE TITLE
Absolute primary decomposition fails for the zero ideal

### ISSUE THREAD

**Issue by simonbrandhorst** [oscar-system/Oscar.jl#2212]
@wdecker 
```
julia> R,(x1,x2) = polynomial_ring(QQ,2);

julia> I = ideal(R,0*x1)
ideal(0)

julia> absolute_primary_decomposition(I)
ERROR: Polynomial must be irreducible
```

**Comment by fieker** [comment]
On Tue, Apr 04, 2023 at 11:56:29PM -0700, simonbrandhorst wrote:
> @wdecker 
> ```
> julia> R,(x1,x2) = polynomial_ring(QQ,2);
> 
> julia> I = ideal(R,0*x1)
> ideal(0)
> 
> julia> absolute_primary_decomposition(I)
> ERROR: Polynomial must be irreducible
> ```
> 
Singular returns the polynomial 1 which will not define a field.

**Comment by wdecker** [comment]
I will change this so that a vector with one entry will be returned:
(ideal(0), ideal(0), ideal(0), 1)
Any objections?

**Comment by thofma** [comment]
Why not return the empty list? Would that also work?

**Comment by simonbrandhorst** [comment]
The zero ideal should be primary because it is prime. Then `(ideal(0), ideal(0), ideal(0), 1)` looks consistent to me?

**Comment by wdecker** [comment]
Yes, a polynomial ring over QQ has no zero-divisors.

**Comment by simonbrandhorst** [comment]
If we want to make it type stable, we could also create a number field `QQ[x]/(x-1)` with
`rationals_as_number_field`. That would also make it easier to use in further code.

**Comment by wdecker** [comment]
resolved in PR #2229
